benchmarks/python/core.py

```diff
@@ -259,33 +260,95 @@ def run_benchmark(
         outputs: Output of the target function
     """
 
-    assert device in [
+    # Check that the device is `cuda` or `host:{compile/steady/dynamic}`.
+    assert device.split(":")[0] in [
         "cuda",
         "host",
-    ], f"Unsupported device type: {device}. Use one of cuda/host."
+    ], f'Unsupported device type: {device.split(":")[0]}. Use one of cuda/host.'
 
-    if device == "host":
+    host_bench_mode = None
+    if device.split(":")[0] == "host":
+        # Host benchmarking expects a fusion function to generate fusion definitions everytime FusionCache is reset.
         assert fusion_fn is not None and benchmark_fn is None
 
-    nvf_benchmark = NVFBenchmark(benchmark, device=device)
+        # Reset the FusionCache to avoid any inadvertent fusion execution from affecting measurements
+        FusionCache.reset()
+
+        # device = 'host:compile', 'host:steady', 'host:dyanamic'
+        # Set the host_bench_mode -- The 3 modes require different setup calls.
+        host_bench_mode = device.split(":")[-1]
+        device = device.split(":")[0]  # device = 'host'
+
+        # Set the warmup rounds if required for `steady/dynamic` host latency measurement.
+        if (
+            host_bench_mode in ["steady", "dynamic"]
+            and BENCHMARK_CONFIG["warmup_rounds"] == 0
+        ):
+            # By default, warmup_rounds=1. If BENCHMARK_CONFIG['warmup_rounds'] == 0 through --benchmark-warmup-rounds, raise a warning that it was ignored.
+            warnings.warn(
+                "--benchmark-warmup-rounds=0 was ignored for host benchmarking. Setting warmup_rounds=1."
+            )
+            BENCHMARK_CONFIG["warmup_rounds"] = 1
```

Comment:
IIUC `BENCHMARK_CONFIG` is like a global configuration shared by multiple benchmark runs. If so, it seems a little odd that it gets modified while executing one instance of benchmarks. What would happen both host and non-host benchmarks were run? Wouldn't that affect non-host benchmarks as well?

Reply:
This was bug, thanks for pointing it out. It has been resolved now. 

Running the following command:
```
pytest benchmarks/python/test_many_pointwise_ops.py benchmarks/python/test_pointwise_mul.py --benchmark-warmup-rounds=0 --benchmark-num-inputs=1 -s -v
```
where `--benchmark-num-inputs=1` is to only run 1 input size for `test_pointwise_mul.py`.

<img width="1089" alt="Screenshot 2024-09-24 at 5 24 40 PM" src="https://github.com/user-attachments/assets/83b105bb-c95e-4cfe-b613-bd50f43c6cfb">

The output shows that the CLI setting for `warmup_rounds` is adhered to for non-host benchmarking in `test_pointwise_mul.py` as well as `host:compile`.
_(Note: The warmup rounds are not automatically printed on stdout, I inserted a print statement for easier debugging and demonstration.)_